BrainPortal/app/controllers/access_profiles_controller.rb

```diff
@@ -0,0 +1,146 @@
+
+#
+# CBRAIN Project
+#
+# Copyright (C) 2008-2012
+# The Royal Institution for the Advancement of Learning
+# McGill University
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+# Simple controller for creating/editing access profiles, which are
+# an admin-only resource.
+class AccessProfilesController < ApplicationController
+
+  Revision_info=CbrainFileRevision[__FILE__] #:nodoc:
+
+  before_filter :login_required
+  before_filter :admin_role_required
+
+  def index #:nodoc:
+
+    @scope = scope_from_session('access_profiles')
+    scope_default_order(@scope, 'name')
+
+    @base_scope       = AccessProfile.where({})
+    @access_profiles  = @scope.apply(@base_scope)
+
+    respond_to do |format|
+      format.html # index.html.erb
+    end
+  end
+
+  def show #:nodoc:
+    @access_profile = AccessProfile.find(params[:id])
+
+    respond_to do |format|
+      format.html # show.html.erb
+    end
+  end
+
+  def new #:nodoc:
+    @access_profile = AccessProfile.new
+  end
+
+  def create #:nodoc:
+    @access_profile = AccessProfile.new(params[:access_profile])
+
+    respond_to do |format|
+      if @access_profile.save
+        flash[:notice] = 'AccessProfile was successfully created.'
+        format.html  { redirect_to :action => :index }
+      else
+        format.html  { render :action  => :new }
+      end
+    end
+  end
+
+  def update #:nodoc:
+    @access_profile = AccessProfile.find(params[:id])
+
+    # Remember 'before' state for the two main associations
+    old_group_ids = @access_profile.group_ids.sort
+    old_user_ids  = @access_profile.user_ids.sort
+
+    # The success variable will be false for errors on ordinary attributes;
+    # for the group_ids and user_ids list, these are always updated with no errors or logging... :-(
+    success = @access_profile.update_attributes_with_logging(params[:access_profile], current_user)
+
+    # Adjust groups and users, and log differences
+    new_group_ids = @access_profile.group_ids.sort
+    new_user_ids  = @access_profile.user_ids.sort
```

Comment:
No. But I like my code to be deterministic. Helps when debugging. The penalty is just 0.000000001 millisecond.